runner/jobs.go

```diff
@@ -25,75 +25,87 @@ type Job struct {
 // - Edit the `run.cns` file
 // - Copy the restraints
 // - Copy the custom toppar
-func (j Job) SetupHaddock24(cmd string) (string, error) {
-
-	logF, err := Run(cmd, j.Path)
-	if err != nil {
-		err := errors.New("Error running HADDOCK: " + err.Error())
-		return logF, err
-	}
-
-	// Edit run.cns
-	runCns := filepath.Join(j.Path, "run1", "run.cns")
-	if err := haddock2.EditRunCns(runCns, j.Params); err != nil {
-		err := errors.New("Error editing run.cns: " + err.Error())
-		return logF, err
+func (j Job) SetupHaddock24(cmd string) error {
+
+	// TODO: Refactor this as a separate function
+	// Append the restraints to run.param
+	m := map[string]string{
+		j.Restraints.Ambig:     "AMBIG_TBL",
+		j.Restraints.Unambig:   "UNAMBIG_TBL",
+		j.Restraints.Hbonds:    "HBOND_FILE",
+		j.Restraints.Dihedrals: "DIHED_FILE",
+		j.Restraints.Hbonds:    "HBOND_FILE",
+		j.Restraints.Tensor:    "TENSOR_FILE",
+		j.Restraints.Cryoem:    "CRYO-EM_FILE",
+		j.Restraints.Rdc1:      "RDC1_FILE",
+		j.Restraints.Rdc2:      "RDC2_FILE",
+		j.Restraints.Rdc3:      "RDC3_FILE",
+		j.Restraints.Rdc4:      "RDC4_FILE",
+		j.Restraints.Rdc5:      "RDC5_FILE",
+		j.Restraints.Rdc6:      "RDC6_FILE",
+		j.Restraints.Rdc7:      "RDC7_FILE",
+		j.Restraints.Rdc8:      "RDC8_FILE",
+		j.Restraints.Rdc9:      "RDC9_FILE",
+		j.Restraints.Rdc10:     "RDC10_FILE",
+		j.Restraints.Dani1:     "DANI1_FILE",
+		j.Restraints.Dani2:     "DANI2_FILE",
+		j.Restraints.Dani3:     "DANI3_FILE",
+		j.Restraints.Dani4:     "DANI4_FILE",
+		j.Restraints.Dani5:     "DANI5_FILE",
+		j.Restraints.Pcs1:      "PCS1_FILE",
+		j.Restraints.Pcs2:      "PCS2_FILE",
+		j.Restraints.Pcs3:      "PCS3_FILE",
+		j.Restraints.Pcs4:      "PCS4_FILE",
+		j.Restraints.Pcs5:      "PCS5_FILE",
+		j.Restraints.Pcs6:      "PCS6_FILE",
+		j.Restraints.Pcs7:      "PCS7_FILE",
+		j.Restraints.Pcs8:      "PCS8_FILE",
+		j.Restraints.Pcs9:      "PCS9_FILE",
+		j.Restraints.Pcs10:     "PCS10_FILE",
```

Comment:
@amjjbonvin did I miss any?

This works by matching the fields in the `yaml` and mapping them to an internal data structure, which then gets passed transformed to the `run.param` in those fields;

```
# input.txt
/some/path/2oob_ti.tbl
/some/path/2oob_hb.tbl
/some/path/2oob_rdcA.tbl
```

```yaml
# config.yaml
scenarios:
      # ...
      restraints:
        ambig: _ti.tbl
        hbonds: _hb.tbl
        rdc1: _rdcA.tbl
```

```
# run.param
AMBIG_TBL=/some/path/2oob_ti.tbl
HBOND_FILE=/some/path/2oob_hb.tbl
RDC1_FILE=/some/path/2oob_rdcA.tbl
```